src/lib/swapper/swappers/utils/helpers/helpers.ts

```diff
@@ -149,33 +109,60 @@ export const getFeesFromFeeData = async ({
   const eip1559Support = await wallet.ethSupportsEIP1559()
 
   if (eip1559Support && maxFeePerGas && maxPriorityFeePerGas)
-    return { gasLimit, maxFeePerGas, maxPriorityFeePerGas }
-  if (gasPrice) return { gasLimit, gasPrice }
+    return {
+      networkFeeCryptoBaseUnit: bn(gasLimit).times(maxFeePerGas).toString(),
+      feesWithGasLimit: { gasLimit, maxFeePerGas, maxPriorityFeePerGas },
+    }
+
+  const displayGasPrice = optimism.isOptimismChainAdapter(adapter)
+    ? (feeDataEstimate as Awaited<ReturnType<optimism.ChainAdapter['getFeeData']>>).l1GasPrice
+    : gasPrice
+
+  if (gasPrice)
```

Comment:
As this is of `string` type, are checking here because there is a change we might have an empty string (`''`)?

Reply:
shouldn't ever happen, but paranoia check

Reply:
i think this was carried over from the original implementation but cant hurt either way